tooling/generate-card/generate-card.js

```diff
@@ -0,0 +1,107 @@
+const fs = require('fs');
+const path = require('path');
+const { createCanvas, loadImage, registerFont } = require('canvas');
+
+registerFont(path.join(__dirname, 'assets', 'coco-regular.ttf'), {
+  family: 'Coco Gothic',
+});
+registerFont(path.join(__dirname, 'assets', 'coco-bold.ttf'), {
+  family: 'Coco Gothic',
+  weight: 'bold',
+});
+
+const DEFAULT_BACKGROUND = path.join(__dirname, 'assets', 'background.png');
+const CARD_WIDTH = 1440;
+const CARD_HEIGHT = 760;
+const PADDING_Y = 126;
+const PADDING_X = 80;
+
+/**
+ * cherry picked from ✨
+ * and modified to curry with context because I love currying to free my mind.
+ * https://codepen.io/peterhry/pen/AGIEa
+ */
+
+const wrapTextFactory = (context) => (text, x, y, maxWidth, lineHeight) => {
+  var words = text.split(' '),
+    line = '',
+    lineCount = 0,
+    i,
+    test,
+    metrics;
+
+  for (i = 0; i < words.length; i++) {
+    test = words[i];
+    metrics = context.measureText(test);
+    while (metrics.width > maxWidth) {
+      // Determine how much of the word will fit
+      test = test.substring(0, test.length - 1);
+      metrics = context.measureText(test);
+    }
+    if (words[i] != test) {
+      words.splice(i + 1, 0, words[i].substr(test.length));
+      words[i] = test;
+    }
+
+    test = line + words[i] + ' ';
+    metrics = context.measureText(test);
+
+    if (metrics.width > maxWidth && i > 0) {
+      context.fillText(line, x, y);
+      line = words[i] + ' ';
+      y += lineHeight;
+      lineCount++;
+    } else {
+      line = test;
+    }
+  }
+
+  context.fillText(line, x, y);
+};
+
+/**
+ * Generate a card image (jpeg format)
+ * by composing a background with a title
+ * and an optional author.
+ */
+async function generateCard({ title, author }, file) {
```

Comment:
that's the core of creating the preview images 👌

Reply:
Straight forward code, simple to read